static/templates/empty_feed_notice.hbs

```diff
@@ -18,6 +18,24 @@
         {{/each}}
     </div>
     {{else}}
+    {{#if is_spectator}}
+        <div class="spectator-empty-narrow-notice">
+            <p>
+                <i class="fa fa-exclamation-circle" aria-hidden="true"></i>
+                {{#tr}}
+                Please note that your experience is limited to
+                <z-link>web public streams</z-link>.
+                {{#*inline "z-link"}}<a href="/help/web-public-streams"
+                target="_blank" rel="noopener noreferrer">{{> @partial-block}}</a>{{/inline}}
+                {{/tr}}
+                &nbsp;
+                {{#tr}}
+                <z-link>Log in</z-link> to access the full experience of Zulip!
+                {{#*inline "z-link"}}<a href="/login/">{{> @partial-block}}</a>{{/inline}}
+                {{/tr}}
+            </p>
+        </div>
+    {{/if}}
```

Comment:
Structurally, I think we don't really want to extend this template; `narrow_banner.js` is responsible for generating the content to put here, and in fact is still producing something in the `{{{ html }}}` block below.

I also notice that the spectator code we have in `narrow_banner.js` is buggy -- it will claim a stream doesn't exist if you're just looking at a topic that doesn't exist.

I'm going to try dropping this commit from this PR; we still want to do it, but I think it'll be easier as a separate PR.